core/embed/rust/src/ui/display/font.rs

```diff
@@ -198,31 +202,28 @@ impl Font {
             return 0;
         }
 
-        if let Some(glyph) = self.get_first_glyph_from_text(text) {
-            glyph.bearing_x
-        } else {
-            0
-        }
+        self.get_first_glyph_from_text(text)
+            .map_or(0, |glyph| glyph.bearing_x)
     }
 
     pub fn char_width(self, ch: char) -> i16 {
-        display::char_width(ch, self.into())
+        let mut buf = [0u8; 4];
+        let encoding = ch.encode_utf8(&mut buf);
+        self.text_width(encoding)
     }
 
     pub fn text_height(self) -> i16 {
-        display::text_height(self.into())
+        display::get_font_info(self.into()).map_or(i16::MAX, |font| font.height.try_into().unwrap())
```

Comment:
i'm thinking, just `unwrap!()` the whole thing, asking for a text_height on an invalid font id should be an unrecoverable error, no? (dtto below)

Reply:
I kept it the same as it is in other equivalent functions in the file. Do you insist on `unwrap!()` instead?

Reply:
yeah, I meant in all of them, not in text_height specifically.
i think `unwrap!` is better here, it seems pointless to spend code on returning value for a nonexistent font?

Reply:
80fa6c254de6a73d4742e3190d0eb8fad00985ba

Reply:
ah but there's still an `unwrap()` in there. we don't want to call `fn unwrap` basically ever, always use `unwrap!`

this should be `unwrap!(display::get_font_info(self.into())).height as i16` (honestly wouldn't worry about doing a try_into ... speaking of, why is it an `int` in the struct and not `uint16_t` ?)

Reply:
I see. `unwrap()` removed.

I kept `int` as it was before. If you don't mind, as changing it to `uint16_t` is a slightly larger change (fonts generator, generated fonts, function headers and callers), I'd postpone it to another PR not to prolong this one.

Reply:
not stopping the PR on it anymore but you could have avoided the map() if you just unwrap!() the result of get-font-info, which would have given you an infallible non-option struct with the height attribute

Reply:
yes, removed the `map` on rebase